dnf/cli/commands/clean.py

```diff
@@ -91,15 +94,29 @@ def set_argparser(parser):
 
     def run(self):
         cachedir = self.base.conf.cachedir
-        types = set(t for c in self.opts.type for t in _CACHE_TYPES[c])
-        files = list(_tree(cachedir))
-        logger.debug(_('Cleaning data: ' + ' '.join(types)))
-
-        if 'expire-cache' in types:
-            expired = _cached_repos(files)
-            self.base._repo_persistor.expired_to_add.update(expired)
-            types.remove('expire-cache')
-
-        patterns = [dnf.repo.CACHE_FILES[t] for t in types]
-        count = _clean(cachedir, _filter(files, patterns))
-        logger.info(P_('%d file removed', '%d files removed', count) % count)
+        md_lock = dnf.lock.build_metadata_lock(cachedir, True)
```

Comment:
I think this should be `dnf.lock.build_metadata_lock(cachedir, self.base.conf.exit_on_lock)`.